### PR TITLE
limit the get of deployments to WATCH_NAMESPACE on sync

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

## Which problem is this PR solving?
- Fixes https://github.com/jaegertracing/jaeger-operator/issues/2085

## Short description of the changes
-  Check scope of the operator and fetch only deployments that are in the WATCH_NAMESPACES if it's not empty
-  If WATCH_NAMESPACES is not empty (non-cluster wide mode) the deployment webhook only process deployments for those namespaces.

### CODE REVIEW

**Review by frzifus — COMMENTED** [review]
N1, would it make sense to add a e2e test to make sure it doesnt break in the future?

**Comment by rubenvp8510** [comment]
> N1, would it make sense to add a e2e test to make sure it doesnt break in the future?

Yes, I'm on it  :)

File: config/namespaced/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ../default
+
+components:
+  - ./patch
```
Comment (frzifus) [inline config/namespaced/kustomization.yaml:8]: missing new line

File: config/namespaced/patch/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+patches:
```
Comment (frzifus) [inline config/namespaced/patch/kustomization.yaml:3]: can you explain the patch? ^^

Comment (frzifus) [reply config/namespaced/patch/kustomization.yaml:3]: Thank you!

**Review by frzifus — APPROVED** [review]

**Review by frzifus — APPROVED** [review]

**Pull request merged**